scripts/bugzilla_attach.py

```diff
@@ -0,0 +1,189 @@
+After doing so, the attachment is encoded as a base64 string.
+In order to authenticate against the Bugzilla website, a username and
+password are prompted. A POST request is then sent to the Bugzilla
+website. If there are any errors (invalid ID or invalid login), the
+script prompts for those and retries the request until there are no
+errors.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+import tarfile
+import datetime
+import base64
+from getpass import getpass
+import requests
+
+MAX_LOGLINES = 100000
+
+BUGZILLA_URL = "https://www.bugzilla.redhat.com"
+
+HEADERS = {'Content-type': 'application/json'}
+
+LOGFOLDER = "gather-files/"
+
+OUTPUT_FILE = "must-gather.log"
+
+ARCHIVE_NAME = "must-gather"
+
+def main():
+    """Main function"""
+
+    # Start with getting command-line argument(s)
+    parser = argparse.ArgumentParser(description="Sends the result of must-gather to Bugzilla.")
+    parser.add_argument("ID", metavar="id", type=int,
+                        help="The ID of the bug in Bugzilla")
+
+    args = parser.parse_args()
+
+    bug_id = args.ID
+
+    # If the log folder already exists, delete it
+    if os.path.isdir(LOGFOLDER):
+        shutil.rmtree(LOGFOLDER)
+
+    # Make a new log folder
+    os.mkdir(LOGFOLDER)
+
+    # Open the output file
+    with open(LOGFOLDER + OUTPUT_FILE, "w+") as out_file:
+        # Run oc adm must-gather with the appropriate image and dest-dir
+        print("Running must-gather")
+        subprocess.run(
+            ["oc", "adm", "must-gather",
+            "--image=quay.io/kubevirt/must-gather", "--dest-dir=" + LOGFOLDER],
+            stdout=out_file)
+    
+    # Recursively walk the log folder
+    print("Searching for files to trim")
+    for subdir, _, files in os.walk(LOGFOLDER):
+        for file in files:
+            file_name = os.path.join(subdir, file)
+            with open(file_name, "r+") as curr_file:
+                # Check the number of lines in each file
+                num_lines = get_lines(curr_file)
+                if num_lines > MAX_LOGLINES:
+                    # If the maximum number of lines is too high, trim it
+                    print("Trimming %s because it exceeds %d lines"
+                        % (os.path.join(subdir, file), MAX_LOGLINES))
+                    trim_file(curr_file, MAX_LOGLINES)
+
+    # Create a time-stamped archive name
+    archive_name = ARCHIVE_NAME + "%s.tar.gz" % datetime.datetime.now().strftime("Y-%m-%d_%H-%M-%S")
+
+    # Add all the files in the log folder to a new archive file
+    with tarfile.open(archive_name, "w:gz") as tar:
+        print("Creating archive: " + archive_name)
+        tar.add(LOGFOLDER, arcname=os.path.basename(LOGFOLDER))
+
+    print("Preparing to send the data to " + BUGZILLA_URL)
+
+    file_data = ""
+    with open(archive_name, "rb") as data_file:
+        file_data = base64.b64encode(data_file.read())
+
+    bugzilla_username = input("Enter Bugzilla username: ")
```

Comment:
that's not ideal if we'd like to use this script in automated env.
please expose required input from environment variables and as input parameters.

Reply:
In that case, it might be better to add an optional parameter that can take an API key (generated from the Bugzilla website), in which case the prompt for username/password would be skipped.

Reply:
Added an optional flag that allows passing an API key instead of prompting for a username and password